### PR TITLE
Feature/a1 mtdc 125 Deploy PRS connectors

### PR DESCRIPTION
**Pull request by ouphi**
- Added consumer helm chart
- Added provider helm chart
- Deploy connectors with helm in terrraform (then it is deployed in deploy workflow)

Green deploy connectors job:
`https://github.com/catenax/tractusx/runs/4163219459?check_suite_focus=true`
Green deploy: `https://github.com/catenax/tractusx/actions/runs/1443449094`
After deploying.
health endpoint available:
`https://catenaxdev001akssrv.germanywestcentral.cloudapp.azure.com/prs-consumer/api/health`

Tested like this:
called
`curl -X POST 'https://catenaxdev001akssrv.germanywestcentral.cloudapp.azure.com/prs-connector-consumer/api/file/test-document?connectorAddress=http://prs-connector-provider/&destination=/tmp/copy/dest/new-document.txt'`
Checked the file appeared in the provider
`kubectl exec -it service/prs-connector-provider -- cat /tmp/copy/dest/new-document.txt`

### CODE REVIEW

File: coreservices/partsrelationshipservice/cd/helm/prs-consumer/values.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+image:
+  repository: prs-connector-consumer
+  tag: "latest"
+ingress:
+  host: consumer.local
+  className: nginx
+  prefix: /prsconsumer
```
Comment (algattik) [inline coreservices/partsrelationshipservice/cd/helm/prs-consumer/values.yaml:7]:
```suggestion
  prefix: /prs-consumer
```

File: coreservices/partsrelationshipservice/cd/helm/prs-consumer/templates/statefulset.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+apiVersion: apps/v1
+kind: StatefulSet
```
Comment (ouphi) [inline coreservices/partsrelationshipservice/cd/helm/prs-consumer/templates/statefulset.yaml:2]: TODO: https://confluence.catena-x.net/display/ARTI/MTPDC+DevOps+and+Infrastructure

**Comment by algattik** [comment]
Worth updating the DevOps ADR to explain the deployment strategy, and use of stateful set.

File: coreservices/partsrelationshipservice/cd/terraform-common/main.tf [hunk 3]
```diff
@@ -8,4 +8,4 @@ module "prs_application_insights" {
   name                = "${var.prefix}-${var.environment}-prs-appi"
   resource_group_name = local.resource_group_name
   location            = local.location
-}
+}
```
Comment (algattik) [inline coreservices/partsrelationshipservice/cd/terraform-common/main.tf:11]: spurious diff

File: coreservices/partsrelationshipservice/cd/terraform-consumer-connector/main.tf [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  set {
+    name  = "ingress.className"
+    value = var.ingress_class_name
+  }
+
+  set {
+    name  = "ingress.prefix"
+    value = "/prs-consumer"
```
Comment (algattik) [inline coreservices/partsrelationshipservice/cd/terraform-consumer-connector/main.tf:30]: prs-connector-consumer

File: coreservices/partsrelationshipservice/cd/terraform-dataspace-partition/README.md [hunk 5]
```diff
@@ -16,11 +17,20 @@ The following tools need to be installed on your system where you run the script
...
+
+You can find several terraform folders in the `cd` folder. To deploy one of them.
+From the Terraform folder that you want to deloy:
+
 1. Sign-on to Azure and select the target subscription for the PoC landscape with `az login --tenant <catenax-tenant>`
-1. Check whether the variables for your target environment have been correctly set `cat variables.tf`
-1. From the main directory of this repository, run `terraform init`
-1. Run `terraform plan`
-1. Run `terraform apply`
+1. Check whether the variables for your target environment have been correctly set `cat terraform-common/variables.tf`
```
Comment (Izzzu) [inline coreservices/partsrelationshipservice/cd/terraform-dataspace-partition/README.md:26]: nit: enumeration typos, should be 1,2,3... ?

Comment (algattik) [reply coreservices/partsrelationshipservice/cd/terraform-dataspace-partition/README.md:26]: Markdown allows both styles. Advantage of using 1,1,1 is auto-numbering (and reduced diffs).

**Review by algattik — APPROVED** [review]

File: coreservices/partsrelationshipservice/cd/terraform-dataspace-partition/README.md [hunk 6]
```diff
@@ -16,11 +17,20 @@ The following tools need to be installed on your system where you run the script
 
 Run the following commands to deploy the PoC infrastructure:
 
+### Deploy terraform
```
Comment (Izzzu) [inline coreservices/partsrelationshipservice/cd/terraform-dataspace-partition/README.md:20]: nit: 
```suggestion
### Deploy with terraform
```
or
```suggestion
### Deploy terraform resources
```
"Deploy terraform" sounds like colloquial term to me, can be misleading, but feel free to ignore if that's not the case

File: coreservices/partsrelationshipservice/cd/terraform-dataspace-partition/README.md (hunk 5 above)
Comment (Izzzu) [reply coreservices/partsrelationshipservice/cd/terraform-dataspace-partition/README.md:26]: cool, good to know, thanks!

**Review by Izzzu — APPROVED** [review]
Approved with some suggestions/questions for the readme doc

**Review by marcgs — APPROVED** [review]

File: coreservices/partsrelationshipservice/cd/helm/prs-connector-provider/templates/statefulset.yaml [hunk 7]
```diff
@@ -0,0 +1,41 @@
+apiVersion: apps/v1
+kind: StatefulSet
```
Comment (marcgs) [inline coreservices/partsrelationshipservice/cd/helm/prs-connector-provider/templates/statefulset.yaml:2]: Is the plan to use a stateless deployment at some point? We then store the connector state externally (using for instance `CosmosTransferProcessStore`

File: coreservices/partsrelationshipservice/cd/terraform-dataspace-partition/README.md (hunk 6 above)
Comment (marcgs) [reply coreservices/partsrelationshipservice/cd/terraform-dataspace-partition/README.md:20]: `provision`is a more accurate wording IMO

File: coreservices/partsrelationshipservice/cd/terraform-dataspace-partition/README.md [hunk 8]
```diff
@@ -16,11 +17,20 @@ The following tools need to be installed on your system where you run the script
...
+
+You can find several terraform folders in the `cd` folder. To deploy one of them.
+From the Terraform folder that you want to deloy:
```
Comment (marcgs) [inline coreservices/partsrelationshipservice/cd/terraform-dataspace-partition/README.md:23]:
```suggestion
To provision terraform execute the following steps from the corresponding folder:
```

File: .github/workflows/partsrelationshipservice-deploy.yml [hunk 9]
```diff
@@ -91,6 +96,26 @@ jobs:
     outputs:
       application_insights_name: ${{ steps.terraform.outputs.application_insights_name }}
 
+  deploy_connectors:
+    needs:
+      - configure
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: coreservices/partsrelationshipservice/cd/terraform-consumer-connector
```
Comment (marcgs) [inline .github/workflows/partsrelationshipservice-deploy.yml:105]: This directory name is not consisten with the naming `prs-connector-consumer` further below. Should we go with `connector-consumer` everywhere?

File: coreservices/partsrelationshipservice/cd/terraform-consumer-connector/main.tf [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  set {
+    name  = "image.tag"
+    value = var.image_tag
+  }
+}
+
+# Deploy the PRS Provider with Helm
+resource "helm_release" "prs-connector-provider" {
```
Comment (marcgs) [inline coreservices/partsrelationshipservice/cd/terraform-consumer-connector/main.tf:45]: We deploy the provided as well here... Is this just temporary?

Comment (ouphi) [reply coreservices/partsrelationshipservice/cd/terraform-consumer-connector/main.tf:45]: Yes it is temporary.
Providers will be deployed in terraform-dataspace-partition as we will have one provider per partitino.

File: .github/workflows/partsrelationshipservice-deploy.yml (hunk 9 above)
Comment (ouphi) [reply .github/workflows/partsrelationshipservice-deploy.yml:105]: Good catch!

File: coreservices/partsrelationshipservice/cd/helm/prs-connector-provider/templates/statefulset.yaml (hunk 7 above)
Comment (ouphi) [reply coreservices/partsrelationshipservice/cd/helm/prs-connector-provider/templates/statefulset.yaml:2]: Ah I don't know good question.
I can add a task to discuss that.

**Pull request merged**